framework/include/systems/NonlinearSystem.h

```diff
@@ -24,17 +24,14 @@
  *
  * It is a part of FEProblemBase ;-)
  */
-class NonlinearSystem : public NonlinearSystemBase,
-                        public libMesh::NonlinearImplicitSystem::ComputePreCheck
```

Comment:
where did the preCheck go? or deleted? I might see it later in the review

Reply:
We don't need it any more. That's one of the very nice changes in this PR. The nonlinear system and its residual correspond to all the degrees of freedom, both interior and face degrees of freedom, instead of previously where I put the interior degrees of freedom in an auxiliary system. Then a user can choose whether they want to statically condense out the interior degrees of freedom in the *preconditioner* (leaving only the face degrees of freedom) or not.

This means a user is also free now to use any SNES algorithm they like as opposed to being limited to line-search without the line search (meaning previously you could only use `-snes_type newtonls -snes_linesearch_type basic`)

Reply:
This is the key class for this change: https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1StaticCondensation.html